replica/database.cc

```diff
@@ -738,6 +738,14 @@ future<> database::parse_system_tables(distributed<service::storage_proxy>& prox
     }));
 }
 
+static auto add_fragmented_listeners(const gms::feature& f, db::commitlog& cl, const db::commitlog::config& ccfg) {
+    return f.when_enabled([&cl, cfg = ccfg]() mutable {
+        if (!std::exchange(cfg.allow_fragmented_entries, true)) {
+            cl.update_configuration(cfg);
```

Comment:
If we want each listener to control one config attribute, they better read-modify-write
the current configuration rather than modify the one captured when the listener is installed
otherwise the changes wouldn't be additive.

Reply:
Sure. Done.